src/view/main_window.py

```diff
@@ -0,0 +1,131 @@
+
+        self.setCentralWidget(self._widgets)
+
+        self._last_cycle_time = [0] * 45
+        self._setup_menubar()
+        self._setup_status_bar()
+
+        self._fish_connector.start()
+        if self._fish_connector:
+            from model.control_desk import set_network_manager
+            set_network_manager(self._fish_connector)
+
+    def _setup_menubar(self) -> None:
+        """Adds a menubar with submenus."""
+        self.setMenuBar(QtWidgets.QMenuBar())
+        menus: dict[str, list[list[str, callable]]] = {"Fish": [["Connect", self._start_connection()],
+                                                                ["Disconnect", self._fish_connector.disconnect],
+                                                                ["Change", self._change_server_name]]
+                                                       }
+        for name, entries in menus.items():
+            menu: QtWidgets.QMenu = QtWidgets.QMenu(name, self.menuBar())
+            self._add_entries_to_menu(menu, entries)
+            self.menuBar().addAction(menu.menuAction())
+
+    def _start_connection(self):  # TODO rework to signals
+        self._fish_connector.start()
+        from model.control_desk import commit_all_bank_sets
+        commit_all_bank_sets()
+
+    def _add_entries_to_menu(self, menu: QtWidgets.QMenu, entries: list[list[str, callable]]) -> None:
+        """ add entries to a menu"""
+        for entry in entries:
+            menu_entry: QtGui.QAction = QtGui.QAction(entry[0], self)
+            menu_entry.triggered.connect(entry[1])
+            menu.addAction(menu_entry)
+
+    def _change_server_name(self) -> None:
+        """change fish socket name"""
+        text, run = QtWidgets.QInputDialog.getText(self, "Server Name", "Enter Server Name:")
+        if run:
+            self._fish_connector.change_server_name(text)
+
+    def _setup_status_bar(self) -> None:
+        """ build status bor"""
+        status_bar = QtWidgets.QStatusBar()
+        status_bar.setMaximumHeight(50)
+        self.setStatusBar(status_bar)
+
+        self.__label_state_update = QtWidgets.QLabel("", status_bar)  # TODO start Value
+        self._broadcaster.connection_state_updated.connect(self._fish_state_update)
+        status_bar.addWidget(self.__label_state_update)
+
+        label_last_error = QtWidgets.QLabel("Error", status_bar)
+        self._fish_connector.status_updated.connect(label_last_error.setText)
+        status_bar.addWidget(label_last_error)
+
+        self._last_cycle_time_widget = QtWidgets.QLabel(str(max(self._last_cycle_time)))
+
+        self._fish_connector.last_cycle_time_update.connect(self._update_last_cycle_time)
+        status_bar.addWidget(self._last_cycle_time_widget)
+
+    def _fish_state_update(self, connected: bool):
+        if connected:
+            self.__label_state_update.setText("Connected")
+        else:
+            self.__label_state_update.setText("Not Connected")
```

Comment:
If the state changed to disconnected we might as well set the visibility of the cycle update time indicator to False.